coral/src/app/components/CoralNavigate.tsx

```diff
@@ -0,0 +1,23 @@
+import { Navigate, NavigateProps } from "react-router-dom";
+
+type Props = Omit<NavigateProps, "to"> & {
+  to: `/${string}`;
+  useLegacy: boolean;
+};
+
+/**
+ * CoralNavigate is a convenience component, which provides unified API
+ * to do navigation either with React Router or by using window.location API
+ *
+ * Once the Reactification is complete, we should migrate all instances of
+ * <CoralNavigate /> to use the React Router <Navigate />
+ */
+function CoralNavigate({ useLegacy, to, ...navigateProps }: Props) {
+  if (useLegacy) {
+    window.location.assign(to);
+    return <></>;
+  }
+  return <Navigate to={to} {...navigateProps} />;
```

Comment:
Nitpick (no action required): To me, the fact that we are using the component instead of the `useNavigate` hook is a little unintuitive. I can see why it was chosen (can be used everywhere regardless of context), but should we be concerned about the divide that may arise between people prefering the hook API to the component one?

Reply:
When to use `const navigate = useNavigate()` vs. ´<Navigate />´ has never been super clear to me. I for some unknown reason prefer to return a component instead of calling a function, which has a side-effect.

Reply:
Agreeing with Mathieu on this one, I also used `navigate()`, mainly because when I use a component I usually expect to see something rendered. 

And strictly speaking (and very pedantic :D ), using `<Navigate />` also basically calls a function with a side effect 🙈

But it's not really important, especially for this PR!

Reply:
If we would call the navigate as following:
```
if (isSuccess) {
  navigate()
  // And nothing gets returned here
}
...
return <... >...</>
```

The rendered component might be unexpected, if the implementor did not explicitly added some logic how it would look after success. I think the `navigate()` would work nicely if we would add `onSuccess` callbacks to the mutations. Otherwise `<Navigate />` works as expected as it does the  side-effect of changing the URL and rendering something (`null`).  ([Reference](https://github.com/remix-run/react-router/blob/main/packages/react-router/lib/components.tsx#L185)).

Reply:
Yes, I think it's easy to find yourself with a component rendering `undefined` for the TS compiler when using `navigate` in this way. It should probably be handled as you said in the `onSuccess` handler of the query, and also should probably use `redirect` instead of `navigate` to be extra explicit https://reactrouter.com/en/main/fetch/redirect

Reply:
Also, if redirecting to the angular app is a one liner, maybe we don't need an abstraction? 🤔 

Here is another approach on my redirect pr: https://github.com/aiven/klaw/pull/507/commits/525b066021f3f52b9b05eb3228adfe28d7f81e17

Reply:
I think your approach is fine and we go can ahead with that one. The purpose for the abstraction was, that if we would remove (or change the type to a literal) the `useLegacy` from the component types, we would see immediately all the occurrences. Now the code is clear, but it will be spread out through the codebase instead of single point of entry :)

Reply:
I think your approach is fine and we go can ahead with that one. The purpose for the abstraction was, that if we would remove (or change the type to a literal) the `useLegacy` from the component types, we would see immediately all the occurrences. Now the code is clear, but it will be spread out through the codebase instead of single point of entry :).

You can open up a PR for that and I will hit the big green button and follow your example on my branch.

Reply:
Yes, I see your point regarding the usefulness of the abstraction, if it is conceived as essentially a placeholder. I really don't mean to bikeshed this to death as we are currently doing ^^'